### PR TITLE
Add spec for multipart duplicate name

### PR DESCRIPTION
**Pull request by arcadiafalcone**
## Why was this change made? 🤔
Handle pattern found in data

## How was this change tested? 🤨

⚡ ⚠ If this change has cross service impact, including data writes to shared file systems, ***run [integration tests](https://github.com/sul-dlss/infrastructure-integration-test)*** and/or test in [stage|qa] environment, in addition to specs. ⚡

### CODE REVIEW

File: spec/services/cocina/mapping/descriptive/mods/name_spec.rb [hunk 1]
```diff
@@ -197,7 +197,7 @@
             {
               name: [
                 {
-                  parallelValue: [
+                  groupedValue: [
```
Comment (ndushay) [inline spec/services/cocina/mapping/descriptive/mods/name_spec.rb:200]: what does this have to do with?

Comment (arcadiafalcone) [reply spec/services/cocina/mapping/descriptive/mods/name_spec.rb:200]: A mistake - it should be changed back to parallelValue.

Comment (ndushay) [reply spec/services/cocina/mapping/descriptive/mods/name_spec.rb:200]: will do.

**Review by ndushay — APPROVED** [review]
